### PR TITLE
Include enumeration information in generated onboard log documentation

### PR DESCRIPTION
**Pull request by peterbarker**
```
.. _ARM:

ARM: Canonical vehicle attitude
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+-----------+----------------------------------------------------------------+
| TimeUS    | microseconds since system startup                              |
+-----------+----------------------------------------------------------------+
| ArmState  | 1 if vehicle armed, 0 otherwise                                |
+-----------+----------------------------------------------------------------+
| ArmChecks | ARMING_CHECKS at time of arming                                |
|           | Bitmask bits:                                                  |
|           | +--------------------------+----+--+                           |
|           | | ARMING_CHECK_ALL         | 0  |  |                           |
|           | +--------------------------+----+--+                           |
|           | | ARMING_CHECK_BARO        | 1  |  |                           |
|           | +--------------------------+----+--+                           |
|           | | ARMING_CHECK_COMPASS     | 2  |  |                           |
|           | +--------------------------+----+--+                           |
.
.
.
|           | | ARMING_CHECK_CAMERA      | 16 |  |                           |
|           | +--------------------------+----+--+                           |
|           | | ARMING_CHECK_AUX_AUTH    | 17 |  |                           |
|           | +--------------------------+----+--+                           |
|           |                                                                |
+-----------+----------------------------------------------------------------+
| Forced    | true if this arming was forced                                 |
+-----------+----------------------------------------------------------------+
| Method    | method used for arming                                         |
|           | Values:                                                        |
|           | +----------------------------+----+--------------------------+ |
|           | | RUDDER                     | 0  |                          | |
|           | +----------------------------+----+--------------------------+ |
|           | | MAVLINK                    | 1  |                          | |
|           | +----------------------------+----+--------------------------+ |
|           | | AUXSWITCH                  | 2  |                          | |
|           | +----------------------------+----+--------------------------+ |
|           | | MOTORTEST                  | 3  |                          | |
|           | +----------------------------+----+--------------------------+ |
|           | | SCRIPTING                  | 4  |                          | |
|           | +----------------------------+----+--------------------------+ |
|           | | TERMINATION                | 5  | only disarm uses this... | |
|           | +----------------------------+----+--------------------------+ |
|           | | CPUFAILSAFE                | 6  | only disarm uses this... | |
|           | +----------------------------+----+--------------------------+ |
```

### CODE REVIEW

**Comment by peterbarker** [comment]
This makes enumerations from the codebase available to be included as part of a LogMessage field's documentation.

The text above shows it in use for the `ARM` message in two forms - one is a bitmask, the other is an enumeration of possible values.

Any comment on the enumeration line in the code is put into the documentation (that's the "only disarm uses" stuff, above).

@Hwurzburg

**Comment by IamPete1** [comment]
Seems like a good idea to me. I was sure we had at least one logged bit field with a single item two bits wide, but I can't find it now. 

For the MP PR we need a bit name, bit description and a bit number/mask.

I think parsing the code is OK but we should have appropriate markup to allow us to give a manual override.

**Comment by peterbarker** [comment]
> 

Maybe add that "manual override" stuff if and when required?

**Comment by peterbarker** [comment]
This PR brings immediate benefits to the Wiki; past this PR our wiki will start to expand the "mode reason" enumeration so someone looking at a log can use the Wiki to work out why their vehicle changed modes *without* having to go and look at the code

**Review by Hwurzburg — APPROVED** [review]
if it looks like whats shown in the first comment, looks good to me!

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1702,14 +1702,6 @@ void GCS_MAVLINK::log_mavlink_stats()
         return;
     }
 
-    enum class Flags {
-        USING_SIGNING = (1<<0),
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:1706]: need to do something sensible for these shifted forms?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
Need to make sure th RHS of the enumeration values come in OK - so cope with both 1,2,4,8 and 1U<<0, 1U<<1 etc

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp:1706]: Done.

Sample output - search for "Auxiliary" in this file:

[LogMessages.rst.zip](https://github.com/ArduPilot/ardupilot/files/10486367/LogMessages.rst.zip)

**Pull request merged**